From dac1d70b0894b993f1e87ff3103a96faa679728b Mon Sep 17 00:00:00 2001
From: Colin Ian King <colin.king@canonical.com>
Date: Mon, 10 Mar 2014 15:29:47 +0000
Subject: [PATCH 25/28] thd_engine.cpp: check fcntl non-blocking error return

fcntl can possibly fail, so it is probably wise to sanity check for
failed O_NONBLOCK fcntl requests.

Signed-off-by: Colin Ian King <colin.king@canonical.com>
---
 src/thd_engine.cpp | 12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git a/src/thd_engine.cpp b/src/thd_engine.cpp
index 842b11f..0db9138 100644
--- a/src/thd_engine.cpp
+++ b/src/thd_engine.cpp
@@ -154,8 +154,16 @@ int cthd_engine::thd_engine_start(bool ignore_cpuid_check) {
 		thd_log_error("Thermal sysfs: pipe creation failed %d:\n", ret);
 		return THD_FATAL_ERROR;
 	}
-	fcntl(wake_fds[0], F_SETFL, O_NONBLOCK);
-	fcntl(wake_fds[1], F_SETFL, O_NONBLOCK);
+	if (fcntl(wake_fds[0], F_SETFL, O_NONBLOCK) < 0) {
+		thd_log_error("Cannot set non-blocking on pipe: %s\n",
+			strerror(errno));
+		return THD_FATAL_ERROR;
+	}
+	if (fcntl(wake_fds[1], F_SETFL, O_NONBLOCK) < 0) {
+		thd_log_error("Cannot set non-blocking on pipe: %s\n",
+			strerror(errno));
+		return THD_FATAL_ERROR;
+	}
 	write_pipe_fd = wake_fds[1];
 	wakeup_fd = THD_NUM_OF_POLL_FDS - 1;
 
-- 
1.9.0

